src/engine/cachingreader.cpp

```diff
@@ -222,7 +222,16 @@ void CachingReader::process() {
     }
 }
 
-SINT CachingReader::read(SINT sample, bool reverse, SINT numSamples, CSAMPLE* buffer) {
+SINT CachingReader::read(SINT start_sample, bool reverse, SINT numSamples, CSAMPLE* buffer) {
```

Comment:
If we revert back to the old version where the interpretation of start_sample already depends on value of reverse (see below) you could keep the ordering of the parameters.